xchange-kraken/src/main/java/org/knowm/xchange/kraken/service/KrakenTradeService.java

```diff
@@ -27,11 +29,15 @@ public KrakenTradeService(Exchange exchange) {
 
   @Override
   public OpenOrders getOpenOrders() throws IOException {
-    return getOpenOrders(createOpenOrdersParams());
+    return getOpenOrders(null);
   }
 
   @Override
   public OpenOrders getOpenOrders(OpenOrdersParams params) throws IOException {
+    if (params != null && params instanceof OpenOrdersParamCurrencyPair) {
+      OpenOrdersParamCurrencyPair openOrdersParamCurrencyPair = (OpenOrdersParamCurrencyPair) params;
+      return KrakenAdapters.adaptOpenOrders(super.getKrakenOpenOrders(), openOrdersParamCurrencyPair.getCurrencyPair());
```

Comment:
adapters should adapt DTO A to DTO B

they should not do client side filtering

the condition you added in the adapter because of this is hard to read and understand

if the exchange does not allow for server side filtering then please to the filtering here, before passing the orders to the adapter

Reply:
Got it. Will do.